src/main/java/org/commcare/formplayer/sandbox/SqlStorageWrapper.java

```diff
@@ -0,0 +1,325 @@
+package org.commcare.formplayer.sandbox;
+
+import org.commcare.formplayer.services.ConnectionHandler;
+import org.javarosa.core.model.condition.RequestAbandonedException;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.util.InvalidIndexException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Vector;
+import java.util.concurrent.Callable;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.datadog.DatadogConfig;
+import io.micrometer.datadog.DatadogMeterRegistry;
+
+/**
+ * @author $|-|!˅@M
+ */
+public class SqlStorageWrapper<T extends Persistable>
+        implements IStorageUtilityIndexed<T>, Iterable<T> {
+
+    private SqlStorage<T> sqliteStorage;
+    private SqlStorage<T> postgresStorage;
+    private DatadogMeterRegistry meterRegistry;
+
+    public SqlStorageWrapper(ConnectionHandler sqliteConnection, ConnectionHandler postgresConnection, Class<T> prototype, String tableName) {
+        sqliteStorage = new SqlStorage<T>(sqliteConnection, prototype, tableName);
+        postgresStorage = new SqlStorage<T>(postgresConnection, prototype, tableName);
+        initMeterRegistry();
+    }
+
+    public void initMeterRegistry() {
```

Comment:
Did you try autowiring existing MeterRegistry into `FormplayerStorageFactory` and passing that in to the wrapper?

```java
    @Autowired
    private MeterRegistry meterRegistry;

...

    @Override
    public IStorageUtilityIndexed newStorage(String name, Class type) {
        return new SqlStorageWrapper(this.sqLiteDB, this.postgresDB, type, name, meterRegistry);
    }
```

Reply:
I did, but it didn't send anything to datadog and I couldn't find any logs in the console as well.

Reply:
I tried again and found that with autowiring, it was logging the metrics with a prefix of `formplayer.` which I guess is why I missed it in the first place.